app/src/main/java/com/example/kevinschweitzer/myapplication/mvp/presenter/CalculatorPresenter.java

```diff
@@ -0,0 +1,114 @@
+package com.example.kevinschweitzer.myapplication.mvp.presenter;
+
+import android.app.Activity;
+import android.util.Log;
+
+import com.example.kevinschweitzer.myapplication.MainActivity;
+import com.example.kevinschweitzer.myapplication.R;
+import com.example.kevinschweitzer.myapplication.mvp.model.CalculatorModel;
+import com.example.kevinschweitzer.myapplication.mvp.view.CalculatorView;
+import com.example.kevinschweitzer.myapplication.util.bus.RxBus;
+import com.example.kevinschweitzer.myapplication.util.bus.obervers.ClearObserver;
+import com.example.kevinschweitzer.myapplication.util.bus.obervers.EqualsObserver;
+import com.example.kevinschweitzer.myapplication.util.bus.obervers.NumberObserver;
+import com.example.kevinschweitzer.myapplication.util.bus.obervers.OperatorObserver;
+import com.example.kevinschweitzer.myapplication.util.operators.Operator;
+
+public class CalculatorPresenter {
+
+    private CalculatorView view;
+    private CalculatorModel model;
+
+    public CalculatorPresenter(CalculatorView view, CalculatorModel model){
+        this.view = view;
+        this.model = model;
+    }
+
+    /**
+     * Save the number clicked in the model and refresh the view
+     * If number 1 to 9 is clicked this action must be done
+     */
+    public void onNumberClicked(String str){
+        model.setNewNumber(str);
+        view.concatNumber(str);
+    }
+
+    /**
+     * Set the operator in the view and take the first number complete
+     * If one operator +, -, *, / is clicked this action must be done
+     */
+    public void onOperatorClicked(Operator operator){
+        model.setOperator(operator);
+        view.concatOperator(operator.getSymbol());
+    }
+
+    /**
+     * Resolve the calculation
+     * If the equals symbol is clicked, this action must be done
+     */
+    public void onEqualsClicked(){
+        Integer result = model.getResult();
```

Comment:
simplify this logic. For example

Activity activity = view.getActivity();
            if(activty==null){
                return;
        }
       Integer result = model.getResult();
            view.setResult(result!=null?result:activity.getResources().getString(R.string.error));

Reply:
No me estaría andando el operador ternario dentro del método. Parece ser porque setResult esta sobrecargado y no me deja en el mismo método retornar un Int o un String